packages/excalidraw/actions/actionHistory.tsx

```diff
@@ -21,7 +21,7 @@ const writeData = (
     !appState.multiElement &&
     !appState.resizingElement &&
     !appState.editingElement &&
-    !appState.draggingElement
+    !appState.newElement
```

Comment:
@Mrazator do you recall if elements being dragged should also prevent writing to history?

Context: we're removing `draggingElement` which wasn't really related to dragging elements, although it was populated with selection element during dragging.

If we need to preserve dragging detection here, we can check `appState.selectedElementsAreBeingDragged`.

Reply:
>  do you recall if elements being dragged should also prevent writing to history?

This condition prevents triggering undo / redo, not preventing "writing to history" (recording) itself. Cases:
1. When elements are being dragged and the undo / redo won't be prevented, triggered undo / redo could be immediately overridden by the dragging element/s, possibly leading to some inconsistent states.
2. A similar case was with preventing undo / redo when there is active selection with selection element.

If we are now decoupling `draggingElement`, we should again consider preventing triggering undo / redo in both of these case.

> elements being dragged should also prevent writing to history

About this specifically, in general, we prevent capturing store increments (~recording history entries) by default and enable it only in certain cases. There isn't any special logic around `draggingElement` which wasn't there already.

One of those things that was there already, is i.e. checking `draggingElement` for scheduling capturing of increments, such as:

<img width="560" alt="Screenshot 2024-08-05 at 12 54 59" src="https://github.com/user-attachments/assets/5b1ac6b6-d0e3-41f4-bb26-a444eabd8650">

Reply:
Thanks @Mrazator ! If I'm understanding you correctly, we should prevent triggering undo/redo, in addition to what's listed above the changed line,
- when creating a new element,
- when selected elements are being dragged,
correct?

Reply:
> when selected elements are being dragged

this is #1

> a similar case was with preventing undo / redo when there is active selection with selection element.

this is #2, having a selection element, which can, but does not have to contain elements (0..*) - selection is also captured in history, and thus triggering undo / redo while selection is active could end in an inconsistent state

> when creating a new element

maybe an additional, third case, but didn't go through the code; not sure how creating a new element is as async action - so that one could trigger an undo / redo during this action; imaging something like resizing, but that one has a separate `appState.resizingElement` entry

Reply:
@Mrazator I can confirm that currently (in this PR) undo/redo while dragging or box-selecting behaves strangely, so I'll disable it to keep parity with production. Thanks for confirm.